docs/source-2.0/guides/client-guidance/identity-auth/index.md

```diff
@@ -0,0 +1,416 @@
+
+    // ...
+
+    return builder.build();
+}
+```
+
+## AuthSchemeResolver
+
+The Smithy IDL allows clients to model operations which support **multiple**
+authentication schemes. Correspondingly, a Smithy client may be loaded with
+multiple AuthSchemes at runtime. The AuthSchemeResolver is the entity through
+which the appropriate AuthScheme is selected and employed for a given operation
+call.
+
+Like the client's EndpointResolver, an AuthSchemeResolver and its inputs are
+typically code-generated for a particular modeled service.
+
+The AuthSchemeResolver accepts data from the request (e.g. request members) and
+client (e.g. SigV4 "region") and returns a list of authentication scheme options
+that the SDK should use when authenticating that request:
+
+```java
+public interface MyServiceAuthSchemeResolver {
+    List<AuthSchemeOption> resolveAuthSchemes(MyServiceAuthSchemeResolverInput input);
+}
+
+public interface AuthSchemeResolverInput {
+    // The single universal input to auth scheme resolution, since Smithy
+    // operations can be modeled to have a specific preference order of AuthSchemes
+    // that differs from the list at the service level.
+	String operation();
+}
+
+public interface MyServiceAuthSchemeResolverInput extends AuthSchemeResolverInput {
+    // The additional parameters exposed by methods here are entirely
+    // unspecified and will vary based on a service's specific needs.
+    //
+    // A SigV4-enabled service, for example, might include a "region" parameter
+    // that specifies the discrete partition of the service (a geographic
+    // location, or perhaps a deployment stage) for which the Smithy client is making
+    // the request.
+}
+```
+
+### AuthSchemeOption
+
+An AuthSchemeOption represents a **possible** AuthScheme that the current
+request may use:
+
+``` java
+public interface AuthSchemeOption {
+    /**
+     * The unique ID of the scheme to use.
+     */
+    String schemeId();
+
+    /**
+     * Opaque container for additional context to be passed to the scheme's
+     * IdentityResolver when retrieving Identity.
+     */
+    Object identityProperties();
+
+    /**
+     * Opaque container for additional context to be passed to the scheme's
+     * Signer when signing the request. These allow the Signer to use
+     * information from the request, client, etc. in the signature generation process.
+     *
+     * For example, a SigV4-enabled service would encode the "region" parameter
+     * into its signer properties.
+     */
+    Object signerProperties();
+}
+```
+
+After calling into the AuthSchemeResolver, the Smithy client should choose the
+first AuthScheme in the resulting list of options that it supports. A scheme is
+considered "supported" if the client is configured with a scheme with its ID
+**and** if the identity resolver for the scheme is available.
+
+This may be implemented like the following:
+
+```java
+public void resolveAuthScheme(OperationContext ctx) {
```

Comment:
Again, including snippets like this seems helpful but we should be consistent. I elected to just model the examples as using this opaque / undescribed "operation context" input that has the stuff you'd expect on a request.